### PR TITLE
Add more tests for compression

### PR DESCRIPTION
**Pull request by akuzm**
Unit tests for different data sequences, and SQL test for float4.

### CODE REVIEW

**Review by svenklemm — APPROVED** [review]

File: tsl/test/src/test_compression.c [hunk 1]
```diff
@@ -319,42 +319,117 @@ test_gorilla_float()
...
+	}
 
-	i = 0;
-	iter =
+	/* Forward decompression. */
+	DecompressionIterator *iter =
 		gorilla_decompression_iterator_from_datum_forward(PointerGetDatum(compressed), FLOAT8OID);
-	for (DecompressResult r = gorilla_decompression_iterator_try_next_forward(iter); !r.is_done;
-		 r = gorilla_decompression_iterator_try_next_forward(iter))
+	ArrowArray *bulk_result =
```
Comment (mkindahl) [inline tsl/test/src/test_compression.c:392]: Are you missing an include here?

Comment (akuzm) [reply tsl/test/src/test_compression.c:392]: Oh, I missed half the files I wanted to include, will fix now.

**Review by mkindahl — APPROVED** [review]
LGTM

File: tsl/test/src/test_compression.c [hunk 2]
```diff
@@ -418,6 +488,99 @@ test_delta2()
...
 
+static void
+test_delta3(bool have_nulls, bool have_random)
+{
+	DeltaDeltaCompressor *compressor = delta_delta_compressor_alloc();
+	Datum compressed;
+
+	int64 values[1015];
+	bool nulls[1015];
+	for (int i = 0; i < 1015; i++)
```
[the remark this answers is not in the document]
Comment (akuzm) [reply tsl/test/src/test_compression.c]: Let's just use a define for all of them.

**Pull request merged**